### ISSUE TITLE
Fix S1117 FN: Support list patterns

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6378]
### Description

FN for S1117 (VariableShadowsField).

Using C#11 list patterns, the rule is not raising an issue.

### Repro steps

```csharp
public class SomeClass
{
    private byte[] somefield;

    public void SomeMethod(byte[] byteArray)
    {
        if (byteArray is [1, 2, 3] somefield) // FN
        {
        }
    }
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
